s2spy/rgdr/rgdr.py

```diff
@@ -79,12 +79,110 @@ def remove_small_area_clusters(ds: XrType, min_area_km2: float) -> XrType:
     valid_clusters = np.array([c for c, a in zip(clusters, areas) if a > min_area_km2])
 
     ds["cluster_labels"] = ds["cluster_labels"].where(
-        np.isin(ds["cluster_labels"], valid_clusters), 0
+        np.isin(ds["cluster_labels"], valid_clusters), "0"
     )
 
     return ds
 
 
+def add_gridcell_area(data: xr.DataArray):
+    """Adds the area of each gridcell (latitude) in km2.
+
+    Note: Assumes an even grid (in degrees)
+
+    Args:
+        data: Data containing lat, lon coordinates in degrees.
+
+    Returns:
+        Input data with an added coordinate "area".
+    """
+    dlat = np.abs(data.latitude.values[1] - data.latitude.values[0])
+    dlon = np.abs(data.longitude.values[1] - data.longitude.values[0])
+    data["area"] = spherical_area(data.latitude, dlat, dlon)
+    return data
+
+
+def assert_clusters_present(data: xr.DataArray) -> None:
+    """Asserts that any (non-'0') clusters are present in the data."""
+
+    if "i_interval" in data.dims:
+        n_clusters = np.zeros(data["i_interval"].size)
+        for i, _ in enumerate(n_clusters):
+            n_clusters[i] = np.unique(data.isel(i_interval=i).cluster_labels).size
+
+        if np.any(n_clusters == 1):  # A single cluster is the '0' (leftovers) cluster.
+            empty_lags = data["i_interval"].values[n_clusters == 0]
```

Comment:
I checked after the loop to avoid users needing to iteratively remove lags from their experiment. 

Let's say the user picks lags 10 through 20, and from lag 15 onward there are no significant clusters. With the current code they would immediately be notified: `No significant clusters found in lag(s): i_interval=[15, 16, 17, 18, 19, 20]`.

Reply:
Good point. On the other hand, if the clustering takes long, and it fails at the first lag, then they won't have to wait for it to complete to be notified of the error. Which scenario is more likely?

Reply:
I really don't know!

When playing with the example data I set `freq='10d'`, and it turns out that there were two "windows of predictability", namely close to the target date `i_interval` = 1:8, and ~7 months earlier: `i_interval` = 21:25.

Having a summary of when significant clusters can be found could be very useful to users, but I am not sure 🤷‍♂️